### PR TITLE
Type matching based on unqualified names enabled in Create.EngineType

### LINKED ISSUE #3385: BHoM_Engine: ExtractAssembly does not capture engine types
#### Description:

...due to an obvious typo `IsOmNamespace` vs `IsEngineNamespace` introduced in https://github.com/BHoM/BHoM_Engine/pull/3341:
https://github.com/BHoM/BHoM_Engine/blob/f1e91381d8918eed3dd173d815f360c0bc9723ae/BHoM_Engine/Compute/ExtractAssembly.cs#L111

FYI @adecler @IsakNaslundBh

### LINKED ISSUE #3386: BHoM_Engine: Create.EngineType only works for matching fully qualified types
#### Description:

Currently `Create.Type` method has the ability to match types based on their unqualified names, i.e. `Type.FullName` rather than `Type.AssemblyQualifiedName`:
https://github.com/BHoM/BHoM_Engine/blob/f1e91381d8918eed3dd173d815f360c0bc9723ae/BHoM_Engine/Create/Type/Type.cs#L57

This is a powerful feature extremely helpful in case of multi-version builds producing many assemblies with same/similar types, e.g. Revit code. The same is needed in `Create.EngineType` to enable versioning of Revit engine assemblies.

FYI @adecler @IsakNaslundBh

### PR DESCRIPTION
**Pull request by pawelbaran**
### Issues addressed by this PR

Closes #3385
Closes #3386

### Test files

Passing `serialisation` and `versioning` checks should be enough of verification.

### Additional comments

Fix for #3385 is almost self-explanatory, while #3386 requires a bit of explanation I believe. The origin of the bug is renaming of Revit assemblies that led me to finding out that `Create.EngineType` matches types based on fully qualified name - after renaming the assemblies the old and new names are different, resulting in a failure. Two steps were required to address that:
1. Enabling matching by unqualified name
2. Enabling picking first type in case of finding multiple matching types, similar to what we did with @IsakNaslundBh in https://github.com/BHoM/BHoM_Engine/pull/3344

### CODE REVIEW

**Comment by pawelbaran** [comment]
@BHoMBot check compliance
@BHoMBot check serialisation
@BHoMBot check versioning

**Comment by pawelbaran** [comment]
@BHoMBot check compliance
@BHoMBot check serialisation
@BHoMBot check versioning

**Comment by pawelbaran** [comment]
@BHoMBot check versioning
@BHoMBot check serialisation

File: BHoM_Engine/Create/Type/EngineType.cs [hunk 1]
```diff
@@ -35,28 +35,36 @@ public static partial class Create
...
+            }
 
-            if (types.Count == 1)
+            if (types.Count == 1 || (takeFirstIfMultiple && types.Count > 1))
                 return types[0];
             else
             {
                 //Unique method not found in list, check if it can be extracted using the system Type
-                Type type = System.Type.GetType(name, silent);
+                Type type = System.Type.GetType(name, false);
```
Comment (adecler) [inline BHoM_Engine/Create/Type/EngineType.cs:67]: @pawelbaran , the changes make sense overall. Just not sure why `silent` was replaced with `false` here.

**Review by IsakNaslundBh — APPROVED** [review]
Happy with changes made here. Approving based on code review rather than testing, relying on the check versioning in particular to flag any potential issues.

**Comment by pawelbaran** [comment]
@BHoMBot check installer
@BHoMBot check core
@BHoMBot check compliance
@BHoMBot check unit-tests

**Comment by pawelbaran** [comment]
@BHoMBot check null-handling

**Review by adecler — APPROVED** [review]
Happy to review based on code inspection and the PR checks passing, especially serialisation and versioning.

**Comment by pawelbaran** [comment]
I am dispensating the failing UT check on the basis of the fact that all reported errors were present in the past (e.g. [here](https://github.com/BHoM/BHoM_Engine/pull/3369/checks?check_run_id=27077776498)). @peterjamesnugent @IsakNaslundBh maybe that's something we could try to clear/fix in the coming sprints?

**Comment by pawelbaran** [comment]
@BHoMBot this is a DevOps instruction. I am authorising dispensation to be granted on check ref. 28350420492

**Comment by pawelbaran** [comment]
@BHoMBot check ready-to-merge

**Pull request closed without merging**